ios/REANodesManager.m

```diff
@@ -552,4 +586,34 @@ - (NSString *)obtainProp:(nonnull NSNumber *)viewTag propName:(nonnull NSString
   return result;
 }
 
+- (void)tryToFlushUpdateBuffer
+{
+  if (_componentUpdateBuffer.count == 0) {
+    return;
+  }
+
+  NSMutableArray *flushedTags = [NSMutableArray new];
+  for (NSNumber *tag in _componentUpdateBuffer) {
+    ComponentUpdate *componentUpdate = _componentUpdateBuffer[tag];
+    if (componentUpdate == Nil) {
+      return;
+    }
+    __weak typeof(self) weakSelf = self;
+    RCTExecuteOnMainQueue(^void() {
```

Comment:
It'd be better if we could get the whole flushing to happen on the UI thread. Now we enqueue each individual command. It also seem like you just add everything to flushedTags – so what's the point of it? Why not just clear _componentUpdateBuffer instead of removing tags from flushedTags?

On top of that it looks like you modify _componentUpdateBuffer from two thread w/o any synchronization. I think that moving flush logic to ui thread would help with this as well

Reply:
I wonder if there is other way than using `uiManagerDidPerformMounting` such that we can get our code to run in the same UI loop. Enqueueing here would only result in changes being reflected on the next frame, no?

Reply:
I packed all updates in one UI block and I moved all modifications of _componentUpdateBuffer to the UI thread.